### DISCUSSION TITLE
Keystore could not be generated for username 'NAME': ca.serialnumberoctetsize must be between 0 and 20 bytes for this serial number generator

### DISCUSSION THREAD

**Discussion by Nathanforet** [Keyfactor/ejbca-ce#234] [Q&A] [answered]
Hello,

I'm once again seeking for some help,
I'm trying to make a new request from my SubCA
![image](https://user-images.githubusercontent.com/50481453/216089474-6d7e8e2a-48fe-4837-8f39-11483b19587c.png)

And during this part I'm filling the name and password (I don't really know what to put here)
![image](https://user-images.githubusercontent.com/50481453/216089754-d1a82189-3696-4c35-90d9-7e449a942df5.png)

And I'm getting this error
![image](https://user-images.githubusercontent.com/50481453/216089111-96acb767-0650-414f-8eba-191cb5647437.png)

What does it mean ?
I'm here if you need more informations.

Thanks.

**Comment by mike-agrenius-kushner** [comment, marked as answer]
Hi, 

What have you set this to in your CA config? See screenshot:

<img width="348" alt="Screenshot 2023-02-01 at 17 48 48" src="https://user-images.githubusercontent.com/26568998/216107992-66f1c67b-6901-41a2-80fe-af5b7193a45c.png">

Cheers,
Mike

**Reply by Nathanforet** [reply]
Oh yes you are right !
I thought I put it on 20 after confirming the CA by the RootCA, it was on -1.

Thank you for the answer !

I have a doubt on what to put in these fields, is it the password from the end entity profiles created before ?
For me this password is used to backup the CA if needed.
I have now this error
![image](https://user-images.githubusercontent.com/50481453/216271209-7d321a5d-b2ec-4465-b9a0-b6b385088756.png)
I've tried to make an end entity profile and put it int the field, it doesn't change the error.

Thanks Mike.

**Reply by mike-agrenius-kushner** [reply]
That error message says that there's a constraint in the EEP that isn't fulfilled, i.e. something set to Required or that doesn't match up to a regex. Check the logs, you should have a more detailed explanation there. 

Cheers,
Mike

**Reply by Nathanforet** [reply]
I needed to turn off this option in the SubCA Certificate:
![image](https://user-images.githubusercontent.com/50481453/216296594-96def1b1-b1ac-49dd-be57-0c8667615186.png)

In the Manage Certification Authorities section

Thanks again Mike !

**Reply by mike-agrenius-kushner** [reply]
Very welcome!
